lib/Runtime/Library/ArrayBuffer.h

```diff
@@ -93,15 +93,14 @@ namespace Js
         virtual BYTE* GetBuffer() const = 0;
         virtual bool IsValidVirtualBufferLength(uint length) const { return false; };
 
-        static bool Is(Var value);
-        static ArrayBufferBase* FromVar(Var value);
-        static ArrayBufferBase* UnsafeFromVar(Var value);
         static int GetIsDetachedOffset() { return offsetof(ArrayBufferBase, isDetached); }
 
     protected:
         Field(bool) isDetached;
     };
 
+    template <> bool VarIs<ArrayBufferBase>(RecyclableObject* obj);
```

Comment:
What's the reason for not providing the implementation of this one template method in the header?

Reply:
I tried in the header and it was a compiler error: the implementation tried to refer to something that wasn't defined yet. Same for any others that I moved to the cpp file.

---
In reply to: [202397007](https://github.com/Microsoft/ChakraCore/pull/5456#discussion_r202397007) [](ancestors = 202397007)